doc/nrf/ug_partition_manager.rst

```diff
@@ -69,18 +69,97 @@ placement: dict{}
       after: list
          Place the partition after the first existing partition in the list.
 
-   The valid values in the lists are `app`, `end`, or the name of any partition.
-   It is not possible to place the partition after `end`.
-
-sub_partitions: list
-   Define sub partitions which span across one or more other partitions.
-   The sub partitions split the total size of the spanned partitions, and therefore have equal size.
-   This property cannot be used together with the `placement` property.
-   When this property is set for a partition, it is required that the `span` property is also set.
+   The valid values in the lists are `app`, `start`, `end`, or the name of any partition.
+   It is not possible to place the partition after `end` or before `start`.
 
 span: list
-   When creating sub_partitions, this option lists what partitions the sub_partitions should span across.
+   This property lists what partitions this partition should span across.
    This property cannot be used together with the `placement` property.
+   Non-existing partitions are ignored.
+   The Partition Manager will fail if a span is empty, or the partitions are not consecutive after processing.
+
+   .. code-block:: yaml
+      :caption: Span example 1 (fixed order, cannot work)
+
+      mcuboot:
+         placement:
+            before: [spm, app]
+
+      spm:
+         placement:
+            before: [app]
+
+      foo:
+         span: [mcuboot, app] # This will fail, because 'spm' will be placed between mcuboot and app.
+
+      # Order: mcuboot, spm, app
+
+   .. code-block:: yaml
+      :caption: Span example 2 (ambigouous order, but should work)
```

Comment:
"Should" ? IMO it should be well defined whether or not a configuration works, maybe just imprecise wording.

Reply:
For complicated configurations, the ambiguity can be much more pronounced, and might be cases where a solution is theoretically possible, but the algorithm won't find it.

I don't want to make ambiguity illegal (too restrictive), and I
also don't want to promise that a solution *always* will be found if it exists (not feasible).

This particular example will work, and indeed "should IMO" work in all future versions of the algorithm (since it is so simple), I just don't want to attempt to draw a line between what is guaranteed and not.

I'll try to find some better wording ;).

Reply:
Take a look now. I added some text and a note before the examples, and changed a comment.